lib/services/devices/deviceService.js

```diff
@@ -337,36 +338,44 @@ function updateRegisterDevice(deviceObj, callback) {
         }
     }
 
-    function extractDeviceDifference(newDevice, oldDevice, callback) {
-        var deviceData = {
-                id: oldDevice.id,
-                name: oldDevice.name,
-                type: oldDevice.type,
-                service: oldDevice.service,
-                subservice: oldDevice.subservice,
-                commands: [],
-                staticAttributes: []
-            },
-            oldActiveKeys,
+    function mixAttributeArrays(oldArray, newArray) {
```

Comment:
Yeah, maybe it's not the best name. GetAttributeDifference may be more appropriate.